openmc/stats/univariate.py

```diff
@@ -156,6 +158,35 @@ def from_xml_element(cls, elem):
         p = params[len(params)//2:]
         return cls(x, p)
 
+    @classmethod
+    def merge(cls, dists, probs):
+        """Merge multiple discrete distributions into a single distribution
+
+        Parameters
+        ----------
+        dists : iterable of openmc.stats.Discrete
+            Discrete distributions to combine
+        probs : iterable of float
+            Probability of each distribution
+
+        Returns
+        -------
+        openmc.stats.Discrete
+            Combined discrete distribution
+
+        """
+        # Combine distributions accounting for duplicate x values
+        x_merged = set()
+        p_merged = defaultdict(float)
+        for dist, p_dist in zip(dists, probs):
```

Comment:
Should we require `dists` and `probs` to be the same length before performing this operation? Should we also require probabilities to sum to 1 in distribution constructors? Or at the least assume `probs` sums to 1 in this function?

Reply:
Oops, I missed this before. Just added a check that the lengths are consistent. As noted in the other comment, the lack of a  check for probabilities summing to 1 is intentional.